src/formatDate.js

```diff
@@ -9,6 +9,41 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const splitedDate = date.split(oldSeparator);
+
+  const newDate = [];
+
+  let yearFormat = toFormat.includes('YY') ? 'YY' : 'YYYY';
+
+  for (let i = 0; i < splitedDate.length; i++) {
+    const oldPart = fromFormat[i];
+    let index = toFormat.indexOf(oldPart);
+
+    if(oldPart.includes('Y') && oldPart !== yearFormat) {
+      index = toFormat.indexOf(yearFormat);
+    }
+
+    const newPart = toFormat[index];
+    
+    if (oldPart.length > newPart.length) {
+      splitedDate[i] = splitedDate[i].split('').slice(-2).join('');
+    }
+
+    if (oldPart.length < newPart.length) {
+      splitedDate[i] =
+        +splitedDate[i] < 30
+          ? `20${splitedDate[i]}`
+          : `19${splitedDate[i]}`;
```

Comment:
When converting a year from 'YY' to 'YYYY', the check for '< 30' to decide the century is correct. However, the conversion logic should be applied only to the year part. Ensure that this logic is applied to the correct part of the date by checking if the current part being processed is indeed the year.